[Feature]: Typescript 5 support

### 🚀 Feature Proposal

It would be great if ts-jest to supports recently released [Typescript 5.0](https://github.com/Microsoft/TypeScript/releases/v5.0.2)

Currently it warns that it doesn't support even Typescript 4.0, so I'm not reporting this as a bug, but rather a feature request:
```
ts-jest[versions] (WARN) Version 5.0.2 of typescript installed has not been tested with ts-jest. If you're experiencing issues, consider using a supported version (>=2.7.0 <4.0.0).
```
and than my tests fail with

```
● Test suite failed to run

    TypeError: ts.getMutableClone is not a function

      at visitor (node_modules/@vue/cli-plugin-unit-jest/node_modules/ts-jest/dist/transformers/hoist-jest.js:63:34)
      at Object.visitNode (node_modules/typescript/lib/typescript.js:85353:21)
      at transformSourceFileOrBundle (node_modules/typescript/lib/typescript.js:87236:51)
      at transformation (node_modules/typescript/lib/typescript.js:108858:16)
      at transformRoot (node_modules/typescript/lib/typescript.js:108881:73)
      at transformNodes (node_modules/typescript/lib/typescript.js:108866:72)
      at emitJsFileOrBundle (node_modules/typescript/lib/typescript.js:109453:[26](https://git.kryptops.io/telltale/frontend/runs/104026?check_suite_focus=true#step:6:27))
      at emitSourceFileOrBundle (node_modules/typescript/lib/typescript.js:109388:7)
      at forEachEmittedFile (node_modules/typescript/lib/typescript.js:109142:26)
```



### Motivation

It would allow allow people to use latest typescript features with ts-jest.

### Example

_No response_

I'm getting the version warning message and then errors like this:
`error TS5097: An import path can only end with a '.ts' extension when 'allowImportingTsExtensions'`

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,12 @@
+# [29.1.0](https://github.com/kulshekhar/ts-jest/compare/v29.0.5...v29.1.0) (2023-03-26)
+
+
+### Features
+
+* Support TypeScript 5.x ([#4064](https://github.com/kulshekhar/ts-jest/issues/4064)) ([db98cc9](https://github.com/kulshekhar/ts-jest/commit/87f27821db99be411288b50a4f9baa7bedb98cc9)), closes [#4048](https://github.com/kulshekhar/ts-jest/issues/4048)
+
+
+
 ## [29.0.5](https://github.com/kulshekhar/ts-jest/compare/v29.0.4...v29.0.5) (2023-01-13)
 
 

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "ts-jest",
-  "version": "29.0.5",
+  "version": "29.1.0",
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
   "bin": {

